### PR TITLE
Including ContainerRemoved in PLEG event reporting

### PR DESCRIPTION
**Pull request by ronnielai**

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -3563,3 +3565,8 @@ func (kl *Kubelet) ListenAndServe(address net.IP, port uint, tlsOptions *server.
 func (kl *Kubelet) ListenAndServeReadOnly(address net.IP, port uint) {
 	server.ListenAndServeKubeletReadOnlyServer(kl, kl.resourceAnalyzer, address, port, kl.containerRuntime)
 }
+
+func isSyncPodWorthy(event *pleg.PodLifecycleEvent) bool {
+	// Filter out events that are not worthy of pod syncing
```
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: nit: move this to the line above the function.
another nit: add a comment saying why we don't sync the pod on this event.

File: pkg/kubelet/pleg/generic.go [hunk 2]
```diff
@@ -134,29 +134,25 @@ func (g *GenericPLEG) Healthy() (bool, error) {
 	return true, nil
 }
 
-func generateEvent(podID types.UID, cid string, oldState, newState plegContainerState) *PodLifecycleEvent {
+func generateEvent(podID types.UID, cid string, oldState, newState plegContainerState) []*PodLifecycleEvent {
```
Comment (yujuhong) [inline pkg/kubelet/pleg/generic.go]: nit: s/generateEvent/generateEvents

The same applies to other functions.

File: pkg/kubelet/pleg/pleg.go [hunk 3]
```diff
@@ -27,9 +27,9 @@ const (
 	ContainerDied    PodLifeCycleEventType = "ContainerDied"
 	// PodSync is used to trigger syncing of a pod when the observed change of
 	// the state of the pod cannot be captured by any single event above.
-	PodSync PodLifeCycleEventType = "PodSync"
-	// Do not use the events below because they are disabled in GenericPLEG.
+	PodSync          PodLifeCycleEventType = "PodSync"
 	ContainerRemoved PodLifeCycleEventType = "ContainerRemoved"
```
Comment (yujuhong) [inline pkg/kubelet/pleg/pleg.go]: nit: move this line above to group it with other container events.

**Comment by yujuhong** [comment]
Looks good with some nits.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (ronnielai) [inline pkg/kubelet/kubelet.go]: done

File: pkg/kubelet/pleg/generic.go (hunk 2 above)
Comment (ronnielai) [inline pkg/kubelet/pleg/generic.go]: done

File: pkg/kubelet/pleg/pleg.go (hunk 3 above)
Comment (ronnielai) [inline pkg/kubelet/pleg/pleg.go]: Isn't that already done?

Comment (ronnielai) [inline pkg/kubelet/pleg/pleg.go]: I think this is already done.

Comment (yujuhong) [inline pkg/kubelet/pleg/pleg.go]: I meant moving this to line 28 if possible :)

Comment (ronnielai) [inline pkg/kubelet/pleg/pleg.go]: Got it. Done

**Pull request merged**